src/Jaeger.Core/Reporters/RemoteReporter.cs

```diff
@@ -1,114 +1,222 @@
 using System;
+using System.Collections.Concurrent;
 using System.Threading;
+using System.Threading.Tasks;
 using Jaeger.Core.Exceptions;
 using Jaeger.Core.Metrics;
-using Jaeger.Core.Transport;
+using Jaeger.Core.Senders;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Logging.Abstractions;
 
 namespace Jaeger.Core.Reporters
 {
-    // TODO: use this to load up spans into a processing queue that will be taken care of by a thread
+    /// <summary>
+    /// <see cref="RemoteReporter"/> buffers spans in memory and sends them out of process using <see cref="ISender"/>.
+    /// </summary>
     public class RemoteReporter : IReporter
     {
-        internal readonly ITransport _transport;
-        internal readonly IMetrics _metrics;
+        public const int DefaultMaxQueueSize = 100;
+        public static readonly TimeSpan DefaultFlushInterval = TimeSpan.FromSeconds(1);
 
+        private readonly BlockingCollection<ICommand> _commandQueue;
+        private readonly Task _queueProcessor;
+        private readonly Timer _flushTimer;
+        private readonly ISender _sender;
+        private readonly IMetrics _metrics;
         private readonly ILogger _logger;
 
-        private RemoteReporter(ITransport transport, ILoggerFactory loggerFactory, IMetrics metrics)
+        internal RemoteReporter(ISender sender, TimeSpan flushInterval, int maxQueueSize,
+            IMetrics metrics, ILoggerFactory loggerFactory)
         {
-            _transport = transport;
-            _logger = loggerFactory.CreateLogger<RemoteReporter>();
+            _sender = sender;
             _metrics = metrics;
+            _logger = loggerFactory.CreateLogger<RemoteReporter>();
+            _commandQueue = new BlockingCollection<ICommand>(maxQueueSize);
+
+            // start a thread to append spans
+            _queueProcessor = Task.Factory.StartNew(ConsumeQueue);
+
+            _flushTimer = new Timer(_ => Flush(), null, flushInterval, flushInterval);
         }
 
-        public async void Dispose()
+        public void Report(Span span)
         {
+            bool added = false;
             try
             {
-                int n = await _transport.CloseAsync(CancellationToken.None).ConfigureAwait(false);
-                _metrics.ReporterSuccess.Inc(n);
+                // Its better to drop spans, than to block here
```

Comment:
It's better to drop spans than to block here (missing apostrophe and comma not needed)